### PR TITLE
[Bugfix][Rust Frontend] Map missing prompt logprobs for single-token prompts in chat and raw generate

### PR DESCRIPTION
**Pull request by FeathBow**
## Purpose

A single-token prompt has no scored prompt-logprobs positions, and the two model runners represent that case differently: Model Runner V1 returns an empty `[0, k+1]` tensor that decodes to zero scored positions, while Model Runner V2 omits the payload entirely (`gpu/sample/prompt_logprob.py` hits `start_idx >= end_idx` after excluding the unscored leading token and never writes the request into `prompt_logprobs_dict`). `/v1/completions` already normalizes the missing payload for a single-token prompt to a leading `[null]` entry (#44938), but `/v1/chat/completions` and `/inference/v1/generate` treat the same condition as an engine failure. A raw generate request with one token id and `prompt_logprobs` set therefore completes generation and then deterministically returns HTTP 500 under Model Runner V2.

This PR applies the completions mapping to both routes: a missing payload for a single-token prompt maps to `[null]`; a missing payload for a multi-token prompt is still a server error. The completions helper moves to `routes/openai/utils/logprobs.rs` (absorbing `decoded_prompt_logprobs_to_maps`, which had become its only caller) and is shared by completions and chat; raw generate keeps its own mapping since it uses a different response shape.

Not a duplicate of #44938: that PR added the prompt-only/single-token contract for `/v1/completions` only; this PR extends the same contract to `/v1/chat/completions` and `/inference/v1/generate`, which still return HTTP 500. No open PR addresses these two routes.

No model evaluation is needed: logits, sampling, and generated tokens are unchanged — the change is response assembly only, and the live A/B below covers it against an unmodified engine. AI assistance was used for this change; the submitter reviewed every changed line and ran the tests below.

## Test Plan

Unit tests and lints:

```
cd rust
cargo nextest run -p vllm-server
cargo clippy -p vllm-server --all-targets -- -D warnings
```

New tests: `collect_generate_maps_prompt_logprobs_for_single_token_prompt` (raw generate: single-token fallback plus multi-token fail-loud) and `prompt_logprobs_maps_reject_missing_multi_token_payload` (the shared helper's fail-loud arm, unreachable through normal collector flows since a healthy engine always sends the multi-token payload).

Live A/B on one GPU (Qwen3-4B, Model Runner V2 selected by default), base b6ff8a2 vs this branch:

```
vllm-rs serve <model> --port 18081 --served-model-name m --max-model-len 512 -- --enforce-eager --gpu-memory-utilization 0.5

curl -s -o resp.json -w 'HTTP:%{http_code}\n' http://127.0.0.1:18081/inference/v1/generate \
    -H 'Content-Type: application/json' \
    -d '{"token_ids":[9707],"sampling_params":{"max_tokens":2,"prompt_logprobs":1}}'
```

Controls on both binaries: the same request with a multi-token prompt, and single-token `/v1/completions` with `prompt_logprobs=1`.

## Test Result

307 tests run, 306 passed, clippy clean (the one failure is a pre-existing TLS handshake-timing test, identical on the unmodified base).

Before (base b6ff8a2):

```
HTTP:500
{"error":{"message":"raw generate response requested prompt_logprobs but generation returned none","type":"server_error","param":null,"code":"server_error"}}
```

After (this branch), same request (response abridged to the relevant fields):

```
HTTP:200
{"choices":[{"index":0,"finish_reason":"length",...}],"prompt_logprobs":[null],...}
```

Controls, unchanged by this branch: multi-token raw generate with `prompt_logprobs=1` returns identical scored payload content on both binaries; single-token `/v1/completions` with `prompt_logprobs=1` returns `"prompt_logprobs":[null]` on both.

### CODE REVIEW

**Comment by BugenZhao** [comment]
@codex review

**Review by BugenZhao — APPROVED** [review]
Thanks!
